### PR TITLE
feat(Other): Add DMA channel number conversion for Zephyr

### PR DESCRIPTION
**Pull request by ttmut**
### Description

Add a DMA channel conversion function to convert chanel numbers between Zephyr and MSDK. For example, if an SoC has two DMA instances named DMA0 and DMA1, with 4 channels each, DMA1 channel numbers should be offset by 4 before passing to MSDK.

| Zephyr DMA0 | MSDK DMA0 |
|--------|--------|
| 0 | 0 |
| 1 | 1 |
| 2 | 2 |
| 3 | 3 |

| Zephyr DMA1 | MSDK DMA1 |
|--------|--------|
| 0 | 4 |
| 1 | 5 |
| 2 | 6 |
| 3 | 7 |

MAX32657 DMA is an exception to this.

### CODE REVIEW

**Review by sihyung-maxim — APPROVED** [review]

**Comment by sihyung-maxim** [comment]
Should I wait until Furkan reviews this PR before merging?

**Review by hfakkiz — APPROVED** [review]

**Comment by ttmut** [comment]
> Should I wait until Furkan reviews this PR before merging?

I guess it can be merged now.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**
